fix: cache metadata by alias

PowderLi opened the pull request:
issue: #34161

1. database --> collectionName --> collectionID && database --> collectionID --> collectionInfo
2. same as shardLeaders
3. expire different metadata in different situations

longjiquan commented on internal/proxy/meta_cache.go:
```diff
@@ -251,19 +249,20 @@ type MetaCache struct {
-	sfDB             conc.Singleflight[*databaseInfo]
+	dbInfo         map[string]*databaseInfo                         // database -> db_info
+	collNameInfo   map[string]map[string]typeutil.UniqueID          // database -> collectionName -> collection_id
+	collInfo       map[string]map[typeutil.UniqueID]*collectionInfo // database -> collectionName -> collection_info
```
database -> collectionID -> collection_info

longjiquan commented on internal/proxy/meta_cache.go:
```diff
@@ -251,19 +249,20 @@ type MetaCache struct {
+	collNameInfo   map[string]map[string]typeutil.UniqueID          // database -> collectionName -> collection_id
+	collInfo       map[string]map[typeutil.UniqueID]*collectionInfo // database -> collectionName -> collection_info
+	collShard      map[string]map[string]typeutil.UniqueID          // database -> collectionName -> collection_id
+	collNameShard  map[string]map[typeutil.UniqueID]*shardLeaders   // database -> collectionName -> collection_leaders
```
Ditto.

longjiquan commented on internal/rootcoord/alter_collection_task.go:
```diff
@@ -58,23 +59,16 @@ func (a *alterCollectionTask) Execute(ctx context.Context) error {
 
-	return redoTask.Execute(ctx)
+	a.Req.CollectionID = oldColl.CollectionID
+	return a.core.broker.BroadcastAlteredCollection(ctx, a.Req)
```
I think we should redo `BroadcastAlteredCollection`.

PowderLi replied:
there is no difference between use `redoTask.AddSyncStep(&BroadcastAlteredCollectionStep`

longjiquan replied:
I mean it shouldn't be a sync step.

longjiquan commented:
/lgtm

PowderLi committed:
cache metadata by alias
1. database --> collectionName --> collectionID && database --> collectionID --> collectionInfo
2. same as shardLeaders
3. expire different metadata in different situations

Signed-off-by: PowderLi <min.li@zilliz.com>

Peter-YoungUk commented:
Is there any reason to close this PR? 
I believe it could cause performance issues when using aliases with search operations, especially under high traffic. 
This is due to cache misses, which lead to every round trip from the proxy to the coordinator.
